mono/mini/branch-opts.c

```diff
@@ -321,6 +321,10 @@ mono_if_conversion (MonoCompile *cfg)
 		bb1 = bb->out_bb [0];
 		bb2 = bb->out_bb [1];
 
+		/* If either bb1 or bb2 is a try block, abort the optimization attempt. */
+		if (bb1->try_start || bb2->try_start)
+			continue;
```

Comment:
nit: Move it above the `restart:` label. The comment in line 307-317 refers to the `if` statement in line 328.

Reply:
Doesn't seem like a big deal? Moving this above the `restart:` label would also require that you supply some sort of bounds check for `out_bb` and initialization for `bb1` and `bb2` above `restart:`.

Alternatively, the comment could be moved instead.

Reply:
Yes, I moved the comments instead.